File: Examples/test-suite/perl5/director_binary_string_rev_len_runme.pl
use strict;
use warnings;
use Test::More tests => 3;
BEGIN { use_ok 'director_binary_string_rev_len' }
require_ok 'director_binary_string_rev_len';

{
  package DirectorReverseLengthStringCallback;
  use base 'director_binary_string_rev_len::Callback';
  sub run {
      my $ret = 0;
      my ($self, $dataBufferAA) = @_;
      if(defined $dataBufferAA) {
        $ret += ord($_) * 2 for(split('', $dataBufferAA));
      }
      return $ret;
  }
}

{
  print "Start\n";
  my $callback = DirectorReverseLengthStringCallback->new();
  my $caller = director_binary_string_rev_len::Caller->new($callback);
  my $sum = $caller->call();

  is($sum, 9*2*8, 'Unexpected sum: ' . $sum);
}

File: Examples/test-suite/octave/director_binary_string_rev_len_runme.m
director_binary_string_rev_len

function self=DirectorReverseLengthStringCallback()
  global director_binary_string_rev_len;
  self=subclass(director_binary_string_rev_len.Callback());
  self.run=@DirectorReverseLengthStringCallback_run;
end
function ret=DirectorReverseLengthStringCallback_run(self, dataBufferAA)
  ret = 0;
  for i = 1:length(dataBufferAA)
    ret = ret + double(dataBufferAA(i)) * 2;
  end
end

callback = DirectorReverseLengthStringCallback();
caller = director_binary_string_rev_len.Caller(callback);
sum = caller.call();

if (sum != 9*2*8)
    error(sum);
end
